src/SConscript

```diff
@@ -635,118 +154,46 @@ if env["PLATFORM"] == 'darwin':
-
-test_utils_sources = Split("""
-    tests/utils/fake_display.cpp
-    tests/utils/fake_event_source.cpp
-    tests/utils/game_config_manager.cpp
-    """)
-
-wesmage_sources = Split("""
-    sdl/utils.cpp
-    sdl/window.cpp
-    tools/dummy_video.cpp
-    tools/exploder_utils.cpp
-    tracer.cpp
-    wesmage/exit.cpp
-    wesmage/filter.cpp
-    wesmage/options.cpp
-    wesmage/wesmage.cpp
-    """)
-client_env.WesnothProgram("wesmage", wesmage_sources + [libwesnoth_core], have_client_prereqs, OBJPREFIX = "wesmage_", LIBS = ["$LIBS", "png"])
+#---campaignd---
+campaignd_sources = GetSources("campaignd")
+env.WesnothProgram("campaignd", campaignd_sources + [libwesnoth_core], have_server_prereqs, OBJPREFIX = "campaignd_")
 
-libtest_utils = test_env.Library("test_utils", test_utils_sources)
-
-test_sources = Split("""
-    tests/gui/fire_event.cpp
-    tests/gui/iterator.cpp
-    tests/gui/test_gui2.cpp
-    tests/gui/test_save_dialog.cpp
-    tests/gui/visitor.cpp
-    tests/main.cpp
-    tests/test_addons.cpp
-    tests/test_commandline_options.cpp
-    tests/test_config.cpp
-    tests/test_filesystem.cpp
-    tests/test_formula_ai.cpp
-    tests/test_formula_core.cpp
-    tests/test_formula_function.cpp
-    tests/test_image_modifications.cpp
-    tests/test_lexical_cast.cpp
-    tests/test_make_enum.cpp
-    tests/test_map_location.cpp
-    tests/test_mp_connect.cpp
-    tests/test_recall_list.cpp
-    tests/test_rng.cpp
-    tests/test_sdl_utils.cpp
-    tests/test_serialization.cpp
-    tests/test_team.cpp
-    tests/test_unit_map.cpp
-    tests/test_util.cpp
-    tests/test_version.cpp
-    tests/test_whiteboard_side_actions.cpp
-    tests/utils/play_scenario.cpp
-    """)
-test_sources.extend(test_env.Object("tests/test_config_cache.cpp"))
-
-test = test_env.WesnothProgram("test", test_sources +  [libtest_utils, libwesnoth_extras, libwesnoth_core, libwesnoth, libwesnoth_sdl, libwesnoth_extras], have_test_prereqs)
-
-create_images_sources = Split("""
-    sdl/utils.cpp
-    sdl/window.cpp
-    tests/create_images.cpp
-    tools/dummy_video.cpp
-    tools/exploder_utils.cpp
-    tracer.cpp
-""")
+#---cutter---
+cutter_sources = GetSources("cutter")
+client_env.WesnothProgram("cutter", cutter_sources + [libwesnoth_core, libwesnoth_sdl, libwesnoth], have_client_prereqs, OBJPREFIX = "cutter_", LIBS = ["$LIBS", "png"])
+
+#---exploder---
+exploder_sources = GetSources("exploder")
+client_env.WesnothProgram("exploder", exploder_sources + [libwesnoth_core, libwesnoth_sdl, libwesnoth], have_client_prereqs, OBJPREFIX = "exploder_", LIBS = ["$LIBS", "png"])
+
+#---test---
+test_sources = GetSources("test")
+test = test_env.WesnothProgram("test", test_sources +  [libwesnoth_extras, libwesnoth_core, libwesnoth, libwesnoth_sdl, libwesnoth_extras], have_test_prereqs)
+#---end valid scons targets---#
+
+#---start invalid scons targets---#
```

Comment:
in the root SConstruct, the `default_target` variable is specified as only accepting the values `wesnoth wesnothd campaignd cutter exploder test`.